src/features/Chat/components/Menu/Item.tsx

```diff
@@ -27,32 +29,35 @@ export const MenuItem = ({ buddy }: Props) => {
     latest,
     isLoading,
   } = useAppSelector(selectBuddyMessages(buddyId));
-  const activeFolder = useAppSelector(state => state.chats.activeFolder);
+  const activeFolder = useAppSelector(selectActiveFolder);
+  const isDefaultFolder = useAppSelector(selectIsDefaultFolder);
+  const isArchived = useAppSelector(selectIsArchivedFolder);
   const activeChat = useAppSelector(selectActiveChat);
-  const defaultChat = useAppSelector(selectDefaultChat);
-  const activeChatId = activeChat?.buddyId ?? defaultChat.buddyId;
+  const isActiveChat = buddyId === activeChat?.buddyId;
 
   const dispatch = useAppDispatch();
   const openChat = () => dispatch(setActiveChat(buddyId));
 
   const getProfileIconColor = () => {
-    if (buddyId === activeChatId) return 'blueDark';
-    if (activeFolder === 'ok') return 'purpleDark';
-    if (activeFolder === 'archived') return 'orangeDark';
+    if (isActiveChat) return 'blueDark';
+    if (isDefaultFolder) return 'purpleDark';
+    if (isArchived) return 'orangeDark';
     return 'redDark';
   };
```

Comment:
hehe, i like the naming more, but this makes me uneasy somehow. but lets not get into this now.
maybe the correct way here is to create a selector for the iconcolor. that also feels somehow wrong though x)

Reply:
Yep I thought about that too.

Reply:
But perfect doesn't exist so yes, let's not get stuck here! 😃